### ISSUE TITLE
e2e Flake: Cluster failed to initialize within 300 seconds (possible metadata server issue?)

### ISSUE THREAD

**Issue by krousey** [kubernetes/kubernetes#20916]
```console
10:12:14 Using master: e2e-scalability-master (external IP: 104.197.128.205)
10:12:14 Waiting up to 300 seconds for cluster initialization.
10:12:14 
10:12:14   This will continually check to see if the API for kubernetes is reachable.
10:12:14   This may time out if there was some uncaught error during start up.
10:12:14 
10:12:14 ......................................................................................................................................................Cluster failed to initialize within 300 seconds.
10:17:15 2016/02/09 10:17:15 e2e.go:309: Error running up: exit status 2
10:17:15 2016/02/09 10:17:15 e2e.go:305: Step 'up' finished in 8m0.913745413s
```

https://cloud.google.com/console/storage/kubernetes-jenkins/logs/kubernetes-e2e-gce-scalability/4496/

**Comment by ihmccreery** [comment]
And also in the PR builder:

https://console.developers.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/20183/kubernetes-pull-build-test-e2e-gce/27684/artifacts/

**Comment by krousey** [comment]
Happened on kubernetes-e2e-gce-slow as well.

https://console.cloud.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gce-slow/2030/

**Comment by krousey** [comment]
Again:

https://console.cloud.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gce-slow/2065/

**Comment by davidopp** [comment]
@mml do you think you could take a look at this?

**Comment by mml** [comment]
@davidopp okie doke.

**Comment by mml** [comment]
I don't see much to debug with, but I'll try to reproduce.

I think when any test fails at this stage, we could at least output whether or not we can ping the master and then include the apiserver logfile if we can find it.  Maybe a stacktrace from the apiserver, too.

**Comment by mikedanese** [comment]
Most generic error in the world. Happened twice this morning in gce-serial

http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-serial/462/
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-serial/464/

**Comment by mml** [comment]
OK.  Mine just failed on the first try.

```
Feb 11 19:54:26 e2e-test-mml-master startupscript: == kubernetes-salt.tar.gz corrupted, sha1 6cf22a7f474695a7b02c4f8e6bbe35b2441c8eb2 doesn't match expected f18d39b707eb8101b18be679d687507cde1d0a60 ==
Feb 11 19:54:26 e2e-test-mml-master startupscript: Validated https://storage.googleapis.com/kubernetes-staging-e71034d0da/e2e-test-mml-devel/kubernetes-salt.tar.gz SHA1 = f18d39b707eb8101b18be679d687507cde1d0a60
Feb 11 19:54:26 e2e-test-mml-master startupscript: Unpacking Salt tree and checking integrity of binary release tar
Feb 11 19:54:26 e2e-test-mml-master startupscript: 
Feb 11 19:54:26 e2e-test-mml-master startupscript: gzip: stdin: not in gzip format
Feb 11 19:54:26 e2e-test-mml-master startupscript: tar: Child returned status 1
Feb 11 19:54:26 e2e-test-mml-master startupscript: tar: Error is not recoverable: exiting now
Feb 11 19:54:41 e2e-test-mml-master startupscript: Couldn't download release. Retrying...
Feb 11 19:54:41 e2e-test-mml-master startupscript: Downloading binary release tar (https://storage.googleapis.com/kubernetes-staging-e71034d0da/e2e-test-mml-devel/kubernetes-server-linux-amd64.tar.gz)
Feb 11 19:54:41 e2e-test-mml-master startupscript:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Feb 11 19:54:41 e2e-test-mml-master startupscript:                                  Dload  Upload   Total   Spent    Left  Speed
Feb 11 19:54:41 e2e-test-mml-master startupscript: #015  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0#015100   127  100   127    0     0   5797      0 --:--:-- --:--:-- --:--:--  6684
Feb 11 19:54:41 e2e-test-mml-master startupscript: == kubernetes-server-linux-amd64.tar.gz corrupted, sha1 6cf22a7f474695a7b02c4f8e6bbe35b2441c8eb2 doesn't match expected 3a0bfc3dd36a357c3c79a203260577937ee9af76 ==
Feb 11 19:54:41 e2e-test-mml-master startupscript: Validated https://storage.googleapis.com/kubernetes-staging-e71034d0da/e2e-test-mml-devel/kubernetes-server-linux-amd64.tar.gz SHA1 = 3a0bfc3dd36a357c3c79a203260577937ee9af76
Feb 11 19:54:41 e2e-test-mml-master startupscript: Downloading Salt tar (https://storage.googleapis.com/kubernetes-staging-e71034d0da/e2e-test-mml-devel/kubernetes-salt.tar.gz)
Feb 11 19:54:41 e2e-test-mml-master startupscript:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Feb 11 19:54:41 e2e-test-mml-master startupscript:                                  Dload  Upload   Total   Spent    Left  Speed
Feb 11 19:54:41 e2e-test-mml-master startupscript: #015  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0#015100   127  100   127    0     0   5075      0 --:--:-- --:--:-- --:--:--  5772
Feb 11 19:54:41 e2e-test-mml-master startupscript: == kubernetes-salt.tar.gz corrupted, sha1 6cf22a7f474695a7b02c4f8e6bbe35b2441c8eb2 doesn't match expected f18d39b707eb8101b18be679d687507cde1d0a60 ==
Feb 11 19:54:41 e2e-test-mml-master startupscript: Validated https://storage.googleapis.com/kubernetes-staging-e71034d0da/e2e-test-mml-devel/kubernetes-salt.tar.gz SHA1 = f18d39b707eb8101b18be679d687507cde1d0a60
Feb 11 19:54:41 e2e-test-mml-master startupscript: Unpacking Salt tree and checking integrity of binary release tar
Feb 11 19:54:41 e2e-test-mml-master startupscript: 
Feb 11 19:54:41 e2e-test-mml-master startupscript: gzip: stdin: not in gzip format
Feb 11 19:54:41 e2e-test-mml-master startupscript: tar: Child returned status 1
Feb 11 19:54:41 e2e-test-mml-master startupscript: tar: Error is not recoverable: exiting now
```

and then

```console
mml@e2e-test-mml-master:~$ curl https://storage.googleapis.com/kubernetes-staging-e71034d0da/e2e-test-mml-devel/kubernetes-salt.tar.gz
<?xml version='1.0' encoding='UTF-8'?><Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message></Error>mml@e2e-test-mml-master:~$ 
```

Two things seem weird:

1. Where's the salt tarball?
2. After the sha1 doesn't match, why does it say "Validated" and then appear to keep going?

**Comment by mml** [comment]
No.2 is obvious:  because that's what the code says to do:  validate and then keep going no matter what was returned.  https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/configure-vm.sh#L571

This is incidental to the flaky test, but I'd like to fix it unless it's on purpose.  I fear if we change this now a bunch of clusters might fail to boot due to hidden bugs in this part of the process, so maybe it's a change to save until after the 1.2 branch is cut.

**Comment by mml** [comment]
OK, so I'm planning to print the contents of /var/log/startupscript.log and /var/log/kube-apiserver.log on failure, and I plan to toggle this on an env var setting: `KUBE_TEST_DEBUG`.  That seems to follow the existing patterns, and I couldn't find an existing env var to reuse.

**Comment by mikedanese** [comment]
Nice detective work. We wouldn't get to the sha validation if we were handling status code >= 400 in download-or-bust. We should retry on these as well. I'm hoping that GCS is just taking a second to converge after pushing the artifact and there aren't other errors we are failing to handle when pushing the artifact.

```
➜  kubernetes git:(controller-manager-componentconfig) curl https://storage.googleapis.com/asdlkfjaskl/lsajdflkdsajfl && echo -e "\n$?"  
<?xml version='1.0' encoding='UTF-8'?><Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist.</Message></Error>
0
```

**Comment by mikedanese** [comment]
Actually that's not what it is... Looks like it is returning a bad error code on 404. I'm wondering why we got past download-or-bust. What is in the file when you cat it after seeing this error?

**Comment by mikedanese** [comment]
Could this have possibly just happened in a GKE cluster?

http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gke-slow/528/console

I'm wondering if these are all the same issue or different

**Comment by mml** [comment]
I don't think the missing tar file is happening elsewhere.  I tracked it down to having switched projects but not removed the *.uploaded.sha1 files.  I doubt that happens in jenkins.

**Comment by mml** [comment]
With a bug inserted and the timeout reduced...

```

  This will continually check to see if the API for kubernetes is reachable.
  This may time out if there was some uncaught error during start up.

..Cluster failed to initialize within 3 seconds.
/var/log/startupscript.log contents:
Warning: Permanently added '104.197.85.192' (ECDSA) to the list of known hosts.
startupscript.log                                                                                                                                                    100%   36KB  35.8KB/s   00:00
Feb 11 22:10:01 e2e-test-mml-master startupscript: Running startup script /var/run/google.startup.script
Feb 11 22:10:01 e2e-test-mml-master startupscript: == kube-up node config starting ==
Feb 11 22:10:01 e2e-test-mml-master startupscript: Networking functional on e2e-test-mml-master (10.240.0.2)
[...]
Feb 11 22:10:32 e2e-test-mml-master startupscript: `/srv/salt-new/reactor' -> `/srv/reactor'
Feb 11 22:10:33 e2e-test-mml-master startupscript: !!!!!!!! DOING SOMETHING BUGGY
Feb 11 22:10:33 e2e-test-mml-master startupscript: Finished running startup script /var/run/google.startup.script
/var/log/kube-apiserver.log contents:
Warning: Permanently added '104.197.85.192' (ECDSA) to the list of known hosts.
scp: /var/log/kube-apiserver.log: No such file or directory
ERROR: (gcloud.compute.copy-files) [/usr/bin/scp] exited with return code [1]. See https://cloud.google.com/compute/docs/troubleshooting#ssherrors for troubleshooting hints.
2016/02/11 14:11:18 e2e.go:309: Error running up: exit status 2
2016/02/11 14:11:18 e2e.go:305: Step 'up' finished in 2m41.007675547s
2016/02/11 14:11:18 e2e.go:141: Error starting e2e cluster. Aborting.
exit status 1
```

**Comment by wojtek-t** [comment]
Three more failures:

gce-slow:
https://storage.googleapis.com/kubernetes-jenkins/logs/kubernetes-e2e-gce-slow/2122/build-log.txt
https://storage.googleapis.com/kubernetes-jenkins/logs/kubernetes-e2e-gce-slow/2123/build-log.txt

gce-scalability:
https://storage.googleapis.com/kubernetes-jenkins/logs/kubernetes-e2e-gce-scalability/4553/build-log.txt

**Comment by mikedanese** [comment]
http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce/11456/console

**Comment by ixdy** [comment]
Not sure if kubemark is too special-snowflake to be useful, but it very recently had a cluster initialization failure which dumped  /var/log/startupscript.log (/var/log/kube-apiserver.log was apparently nonexistent?):
https://storage.cloud.google.com/kubernetes-jenkins/logs/kubernetes-kubemark-gce/5593

**Comment by gmarek** [comment]
Yeah, I need to add a kubelet on the master machine. I'll try to do this next week.

**Comment by ixdy** [comment]
Both http://ci-test.k8s.io/kubernetes-e2e-gce/11665 and http://ci-test.k8s.io/kubernetes-e2e-gce/11566 recently filed to start the cluster.

**Comment by ihmccreery** [comment]
Is this possibly related to #21322?

**Comment by mml** [comment]
I can see some problems in the startupscript log from the failed runs @ixdy linked, but it's unclear if those are normal or not, since we don't save logs on success.  I'll do a run from my workstation and look for differences.

**Comment by mml** [comment]
I ran from head, so the startupscript is a bit different, but either way all the "error" messages are present in both successful and failed runs.

So that leaves us with: the startupscript seems to run to completion, but the apiserver never gets far enough to even create a 0-byte logfile.

Are there any newer failures?

**Comment by ixdy** [comment]
@mml one from last night: https://storage.cloud.google.com/kubernetes-jenkins/logs/kubernetes-e2e-gce/11801

**Comment by mikedanese** [comment]
How can I add /var/log/startupscript.log to the artifacts dir?

**Comment by ixdy** [comment]
@mikedanese for regular e2e runs where the cluster successfully starts up, you mean? `test/e2e/core.go`.

**Comment by mml** [comment]
I can't find anything useful in startupscript.log.  Talked to @mikedanese and he suggests adding the docker and kubelet logs, which I'll do now.

**Comment by bgrant0607** [comment]
Just encountered this again.
Buckets/kubernetes-jenkins/pr-logs/pull/21576/kubernetes-pull-build-test-e2e-gce/29449

**Comment by bprashanth** [comment]
http://kubekins.dls.corp.google.com:8081/job/kubernetes-pull-build-test-e2e-gce/29659/console
https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/21431/kubernetes-pull-build-test-e2e-gce/29659/

From docker logs:
```
time="2016-02-21T21:17:17.765972579Z" level=fatal msg="Error starting daemon: Error initializing 
network controller: Error creating default \"bridge\" network: failed to allocate gateway (10.246.0.1): No 
available addresses on this pool" 
time="2016-02-21T21:17:32.890888199Z" level=warning msg="Could not get list of networks during 
endpoint cleanup: could not find endpoint count key 
docker/network/v1.0/endpoint_count/f3e4c196f40ac9e727b3198ebd24a400926124e6e7846a79277a9
79088253041/ for network bridge while listing: Key not found in store" 
```

Which smells like https://github.com/kubernetes/kubernetes/issues/20995
@kubernetes/goog-node IIUC we're running 1.9 in the GCI image now, right?

**Comment by dchen1107** [comment]
Yes, it looks like #21523 caused by unclean shutdown docker.  https://github.com/kubernetes/kubernetes/pull/21703 was merged a couple of minutes ago, which should fixed / mitigate hugely on this issue.

**Comment by bprashanth** [comment]
This happened once last night on https://github.com/kubernetes/kubernetes/pull/21706#issuecomment-187456964 but the lack of logs makes it impossible to debug

**Comment by mml** [comment]
Wow, we are really playing whack-a-mole with collecting the logs of interest in every possible case.  Is there a bug somewhere to make it easier to specify "here's a test: always collect these files, and when it fails additionally collect these files".

**Comment by bprashanth** [comment]
>Wow, we are really playing whack-a-mole with collecting the logs of interest in every possible case. Is there a bug somewhere to make it easier to specify "here's a test: always collect these files, and when it fails additionally collect these files".

Just the normal system logs (i.e apiserver, controllermanager, kubelet, docker etc that we collect for every test failure) should help: https://github.com/kubernetes/kubernetes/issues/21722. If we can't ssh/rsync logs from some nodes, that's useful to know as well.

**Comment by dchen1107** [comment]
Quickly scan through the limited available logs with @bprashanth. Based on the timestamp, I think the test failure could be fixed by one of our prs fixing node bootup and being merged yesterday after 17:00pm:

1) #21641: mitigate the issue caused by hairpin mode
2) #21703: gracefully shutdown docker daemon
3) #21727: remove potential corrupted docker network checkpoint file
before start docker daemon.

```
Detected 5 ready nodes, found 6 nodes out of expected 6. Your cluster may not be fully functional.
NAME                              STATUS     AGE
e2e-gce-builder-4-1-minion-3mfi   Ready      25m
e2e-gce-builder-4-1-minion-7s3g   Ready      25m
e2e-gce-builder-4-1-minion-jspw   NotReady   25m
e2e-gce-builder-4-1-minion-pjfa   Ready      25m
e2e-gce-builder-4-1-minion-pm1y   Ready      25m
e2e-gce-builder-4-1-minion-we5c   Ready      25m
2016/02/22 16:53:10 e2e.go:278: Error running up: exit status 1
2016/02/22 16:53:10 e2e.go:274: Step 'up' finished in 28m16.303612093s
2016/02/22 16:53:10 e2e.go:140: Error starting e2e cluster. Aborting.
exit status 1
Build step 'Execute shell' marked build as failure
Recording test results
ERROR: Step ‘Publish JUnit test result report’ failed: No test report files were found. Configuration error?
[PostBuildScript] - Execution post build scripts.
[kubernetes-pull-build-test-e2e-gce] $ /bin/bash -xe /tmp/hudson4488828338393054550.sh
+ [[ -x ./hack/jenkins/upload-to-gcs.sh ]]
+ ./hack/jenkins/upload-to-gcs.sh
Uploading to gs://kubernetes-jenkins/pr-logs/pull/21706/kubernetes-pull-build-test-e2e-gce/29870 (attempt 1)
```

Of course, if we have logs, we can quickly confirm the problem.

**Comment by gmarek** [comment]
There's a long standing p0 for collecting those logs: #10423

**Comment by dchen1107** [comment]
FYI: I filed a separate issue related to node bootup: #22071

**Comment by caesarxuchao** [comment]
FWIW, another occurrence: https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/22156/kubernetes-pull-build-test-e2e-gce/30907/

**Comment by bprashanth** [comment]
Looks like the master kubelet was just stalling trying to read `instance/attributes/google-container-manifest`. From startup logs:
https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/22156/kubernetes-pull-build-test-e2e-gce/30907/artifacts/
```
Feb 29 03:26:57 e2e-gce-master-1-master startupscript: Running startup script /var/run/google.startup.script
Feb 29 03:26:57 e2e-gce-master-1-master startupscript: == kube-up node config starting ==
Feb 29 03:28:37 e2e-gce-master-1-master startupscript: Networking functional on e2e-gce-master-1-master (10.240.0.2)
```
This step usually takes seconds. From kubelet logs, there's a 5m lapse:
```
I0229 03:26:59.261229    2684 kubelet.go:1960] Starting kubelet main sync loop.
W0229 03:27:04.214933    2684 http.go:60] Failed to read pods from URL: http://metadata.google.internal/computeMetadata/v1/instance/attributes/google-container-manifest: 404 Not Found
W0229 03:27:24.216222    2684 http.go:60] Failed to read pods from URL: http://metadata.google.internal/computeMetadata/v1/instance/attributes/google-container-manifest: 404 Not Found
W0229 03:27:44.217967    2684 http.go:60] Failed to read pods from URL: http://metadata.google.internal/computeMetadata/v1/instance/attributes/google-container-manifest: 404 Not Found
W0229 03:28:04.219400    2684 http.go:62] Failed to read pods from URL. Won't log this message anymore: http://metadata.google.internal/computeMetadata/v1/instance/attributes/google-container-manifest: 404 Not Found
E0229 03:32:19.317409    2684 kubelet.go:1361] Mirror pod not available
```

The apiserver only comes up 6m after the startup scripts is invoked. 
```
Flag --address has been deprecated, see --insecure-bind-address instead
I0229 03:32:51.540865       7 server.go:188] Will report 10.240.0.2 as public IP address.
```

**Comment by mml** [comment]
There is bound to be a long tail of startup latencies.

One way to minimize the noise from it is to always retry a suite *once* if the failure is due to a timeout waiting on an async process.

**Comment by bprashanth** [comment]
I agree with that if it's really something like 99%ile and the reason is not something lame that we can fix. https://github.com/kubernetes/kubernetes/pull/22192 adds some logging and actually fixes an unrelated slow http client/connection leak.

**Comment by davidopp** [comment]
Any thoughts on next steps for this issue? (It's marked P0 v1.2).

It doesn't sound like we've identified an actual code bug here. The fix for #10423 has been merged, so we can get better logs in the future. I'd propose we close this issue for now. Any objections?

**Comment by bprashanth** [comment]
we *just* merged: https://github.com/kubernetes/kubernetes/pull/22192, which is the fallout from the debugging in https://github.com/kubernetes/kubernetes/issues/20916#issuecomment-190331119. I want to see why/if/for how long the metadata server is hanging. Previously we weren't timing out requests, so there's a chance we won't hit that failure mode if it was a GET hang of some sort that came after the string of 404s.

**Comment by bprashanth** [comment]
I'm reassigning to myself so I can take a look with the next failure.

**Comment by davidopp** [comment]
OK, but can you re-title this issue so it's just about trying to read from the metadata server?

**Comment by bprashanth** [comment]
Hmm, but I don't even know if that's the problem

**Comment by bprashanth** [comment]
This happened again in http://kubekins.dls.corp.google.com:8081/job/kubernetes-pull-build-test-e2e-gce/32023/console and it looks like we just didn't create the master. 

```
INSTANCE_GROUPS=e2e-gce-builder-2-0-minion-group
NODE_NAMES=e2e-gce-builder-2-0-minion-0501 e2e-gce-builder-2-0-minion-4pdi e2e-gce-builder-2-0-minion-ekcs e2e-gce-builder-2-0-minion-gj5m e2e-gce-builder-2-0-minion-h9co e2e-gce-builder-2-0-minion-pv6j
ERROR: (gcloud.compute.instances.describe) Could not fetch resource:
 - The resource 'projects/kubernetes-jenkins-pull/zones/us-central1-f/instances/e2e-gce-builder-2-0-master' was not found
2016/03/07 10:20:02 e2e.go:200: Error running up: exit status 1
2016/03/07 10:20:02 e2e.go:196: Step 'up' finished in 7m42.604863565s
2016/03/07 10:20:02 e2e.go:110: Error starting e2e cluster. Aborting.
```

I'm spinning that off into another bug.

**Comment by goltermann** [comment]
@bprashanth what remains to close this bug?  Should it be closed and we only track #22655?

**Comment by bprashanth** [comment]
The last issue that Wojtek posted logs too seems to be that etcd wasn't up, filed:
https://github.com/kubernetes/kubernetes/issues/22819

I haven't observed the metadata server issue that I'm keeping this open for since I put in the poll+timeout, so I'm closing this for now as suggested by David in https://github.com/kubernetes/kubernetes/issues/20916#issuecomment-192033284. Please re-open if it reoccurs, and we can dig through logs and triage.

**Comment by wojtek-t** [comment]
This just happened again:
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/22682/kubernetes-pull-build-test-e2e-gce/32360/build-log.txt

It seems that master components, didn't event started, although we have some logs from master kubelet:
it seems like some metadata-serve issue, but I'm not sure.

**Comment by bprashanth** [comment]
That's really strange. The test ran 9 hours ago: https://github.com/kubernetes/kubernetes/pull/22682#issuecomment-195285625

And the kubelet logs have:
```
W0311 09:18:34.803067    2661 http.go:62] Failed to read pods from URL. Won't log this message anymore: http://metadata.google.internal/computeMetadata/v1/instance/attributes/google-container-manifest: 404 Not Found
```

Which I removed in https://github.com/kubernetes/kubernetes/pull/22192/files#diff-d983c6311a6046434cdbf5d6edfc84f9R66 and went in a week ago. So I'm assuming the run didn't have my timeout fix as well, from the same pr. 

@kubernetes/goog-testing am I missing something or is the builder confused?

**Comment by mikedanese** [comment]
Maybe it was an older PR that had not been rebased recently. Do we attempt to merge with HEAD before building the test artifact like travis does?

**Comment by wojtek-t** [comment]
Actually, I agree with @bprashanth - there is something extremely strange here.
@kubernetes/goog-testing - any thoughs on it?

**Comment by spxtr** [comment]
@mikedanese Yes, PR Jenkins checks out the PR merge tag when possible. If the PR has a merge conflict then we will get whatever the user has in their branch, which may be very out of date. We should probably disallow building/testing PRs with merge conflicts.

**Comment by ixdy** [comment]
The build log shows no merge conflicts in that build, and indeed shows
```
Checking out Revision cbb2d1a868465d30c8cb61f4bfcabb7cf8a61eb4 (refs/remotes/origin/pr/22682/merge)
```

I really don't know what happened.

**Comment by bprashanth** [comment]
Im putting the reason this was re-opened down to: infra weirdness.
Out of the last 2 prs that were linked against this:

https://github.com/kubernetes/kubernetes/issues/22933 failed because of:
```
e2e-gce-master-1-master-https e2e-gce-master-1 0.0.0.0/0  tcp:443          e2e-gce-master-1-master
ERROR: (gcloud.compute.addresses.create) Some requests did not succeed:
 - Backend Error
Attempt 1 failed to create static ip e2e-gce-master-1-master-ip. Retrying.
ERROR: (gcloud.compute.addresses.create) Some requests did not succeed:
 - The resource 'projects/kubernetes-jenkins-pull/regions/us-central1/addresses/e2e-gce-master-1-master-ip' already exists
Attempt 2 failed to create static ip e2e-gce-master-1-master-ip. Retrying.
ERROR: (gcloud.compute.addresses.create) Some requests did not succeed:
```
which has an associated internal bug, and is not this issue.

This last failure is the only one concerning, for which I've forked https://github.com/kubernetes/kubernetes/issues/23967
